site/content/entry/yuzu-progress-report-feb-2021/index.md

```diff
@@ -0,0 +1,223 @@
+This is known as `GPGPU` - [General-purpose computing on graphics processing units](https://en.wikipedia.org/wiki/General-purpose_computing_on_graphics_processing_units) - and it's intended to be used when there is a problem that can be separated into a number of parallel tasks in order to be processed more efficiently. These problems are commonly called `embarrassingly parallel problems`.
+
+One of these cases was the [use of compute shaders to decode ASTC textures](https://github.com/yuzu-emu/yuzu/pull/5927).
+`ASTC` stands for "Adaptable Scalable Texture Compression," and it's a fairly new image compression format mainly aimed at mobile devices.
+The Nintendo Switch is capable of decoding these textures natively in hardware, but it's a feature that most PC GPU vendors lack in their products (with the exception of Intel Graphics, being the only vendor that offers native support).
+The decoding of these textures is therefore a non-trivial task that can have a huge impact on performance, as seen in games such as `Astral Chain` and `Luigi's Mansion 3`.
+
+{{< imgs
+	"./astral_chain_atsc.mp4| Comparison between the old and the new implementation of the ASTC decoder"
+  >}}
+
+This led to the implementation of an `ASTC` decoder through the CPU, which was faster than what GPUs could do with their lack of support, but was still far from being a satisfactory solution since it consumed CPU resources and consequently slowed down games that made extensive use of this format.
+The solution, thus, was to implement the decoding through compute shaders.
+Since this is an embarrassingly parallel process, it's more fit to be performed on the GPU by manipulating the data through `GPGPU`.
+This way, the load on the CPU will be shifted to the GPU, allowing emulation to run in parallel with the texture decoding.
+As a side benefit, now textures remain in the GPU memory all the time, since they don't need to be transferred between CPU and GPU for decoding.
+This means that there won't be time spent downloading the texture to the CPU and then uploading it back to the GPU after the decoding is done, like in the old implementation.
+
+This feature works as intended on all GPU vendors on Windows, although there are a few problems on Linux (more specifically, the `AMDGPU-PRO` driver) that still need to be ironed out.
+Our devs are working hard to solve these bugs, so we ask our tux-friends to be patient and stay tuned!
+
+Since compute programs were originally meant to manipulate image data, they also worked out nicely to fix a problem with one of the rendering APIs used in yuzu: by [using compute shaders to swizzle BGR textures on copy](https://github.com/yuzu-emu/yuzu/pull/5891).
+
+{{< single-title-imgs
+    "Color-swapped and properly swizzled versions of Octopath Traveler's title screen"
+    "./octopath1.jpg"
+    "./octopath2.jpg"
+  >}}
+
+In OpenGL, colours are stored in channels, and the way they are laid out varies depending on the format used.
```

Comment:
```suggestion
In OpenGL, colors are stored in channels, and the way they are laid out varies depending on the format used.
```
are you a brit?

Reply:
Will probably be needed to find the other references and change them too.